tests/views/contact.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from __future__ import absolute_import
+
+import pytest
+
+from django.urls import reverse
+
+from pootle_store.models import Unit
+
+
+@pytest.mark.django_db
+def test_contact_report_form_view(client, admin, rf):
+    unit = Unit.objects.last()
+    url = '%s?report=%s' % (reverse('pootle-contact-report-error'), unit.pk)
+    client.login(username=admin.username, password=admin.password)
+    response = client.get(url)
+
+    assert 'contact_form_title' in response.context
+    url = reverse('pootle-contact-report-error')
+    assert response.context['contact_form_url'] == url
+
+    with pytest.raises(AttributeError):
+        assert response.context['form'].unit == unit
+        assert response.context['view'].unit == unit
+
+
+@pytest.mark.django_db
+def test_contact_report_form_view_no_unit(client, admin, rf):
+    url = reverse('pootle-contact-report-error')
+    client.login(username=admin.username, password=admin.password)
+    response = client.get(url)
+
+    with pytest.raises(AttributeError):
+        assert response.context['view'].unit is None
+    assert 'context' not in response.context['view'].get_initial()
+
+
+@pytest.mark.django_db
+def test_contact_report_form_view_blank_unit(client, admin, rf):
+    url = '%s?report=' % reverse('pootle-contact-report-error')
+    client.login(username=admin.username, password=admin.password)
```

Comment:
with view tests its generally a good idea to use the `request_users` fixture to test for each